runsc/container/container.go

```diff
@@ -287,65 +289,80 @@ func New(conf *config.Config, args Args) (*Container, error) {
 		if err != nil {
 			return nil, fmt.Errorf("error creating pod mount hints: %w", err)
 		}
-		goferFilestores, goferConfs, err := c.createGoferFilestores(conf.GetOverlay2(), mountHints)
+		rootfsHint, err := boot.NewRootfsHint(args.Spec)
+		if err != nil {
+			return nil, fmt.Errorf("error creating rootfs hint: %w", err)
+		}
+		goferFilestores, goferConfs, err := c.createGoferFilestores(conf.GetOverlay2(), mountHints, rootfsHint)
 		if err != nil {
 			return nil, err
 		}
+		if !goferConfs[0].ShouldUseLisafs() && conf.NVProxyDocker {
+			return nil, fmt.Errorf("--nvproxy-docker cannot be used together with non-LISAFS backed root mount")
+		}
 		c.GoferMountConfs = goferConfs
 		if err := nvProxyPreGoferHostSetup(args.Spec, conf); err != nil {
 			return nil, err
 		}
-		if err := runInCgroup(containerCgroup, func() error {
-			ioFiles, specFile, err := c.createGoferProcess(args.Spec, conf, args.BundleDir, args.Attached)
+
+		var (
+			ioFiles    []*os.File
+			mountsFile *os.File
+		)
+		if c.shouldSpawnGofer() {
+			if err := runInCgroup(containerCgroup, func() error {
```

Comment:
But now we are not running the sandbox process inside the cgroup. (`sandbox.New(conf, sandArgs)` needs to be called from instead `runInCgroup()`)

We can just have the `c.shouldSpawnGofer()` check inside `c.createGoferProcess()`? And return `c.prepareMounts(rootfsHint)` if gofer process does not have to be created? I think it is much cleaner and allows us to preserve much of the code structure. Also that way we don't need to update both callers of `createGoferProcess()`.

Reply:
Oops! Will fix it. And it does make more sense to do this inside `c.createGoferProcess()`. Thanks! (I was trying to avoid the overhead of joining the cgroups when there is no need to create the gofer process, but I totally forgot the sandbox process created in `sandbox.New()`..)